### DISCUSSION TITLE
Android `IAP.requestPurchase()` suddenly says product not found

### DISCUSSION THREAD

**Discussion by Mkassabov** [hyochan/react-native-iap#1378] [Q&A] [answered]
I want to preface this by saying that at one point everything was working perfectly fine (this was a few months ago). Since then the code hasn't been changed, nor has the library version been upgraded.

What used ot happen is that `IAP.requestPurchase()` would go through without an issue, now, regardless of the product it throws and error that `[Error: That item is unavailable.]`

I have license testers, there are products with the same ID in the actual play console. `getProducts()` works without an issue.

I'm really lost here, any ideas?

**Comment by andresesfm** [comment]
What was the last version that worked for you? Is this for iOS or Android?
Have you tried the latest 7.x?

**Reply by haoflynet** [reply]
Same issue for version 4.x and 8.x, works on ios, but error on android

I can get all active products using `RNIap.getProducts`, but when I `RNIap.requestPurchase(productId)`, it throws the error without any pop-up payment page.

Btw, the Gmail address of my device is added into `License Testing`, and I'm testing it by Android Studio -> Real Device without uploading it to google play console.

@Mkassabov @andresesfm Could you help me?

**Comment by ram-vilash** [comment, marked as answer]
Upload the apk play console in alpha mode then you will be able to test that

**Reply by haoflynet** [reply]
It works now, it seems that I have to wait more hours after the release is approved by Google
